org.eclipse.zest.core/src/org/eclipse/zest/core/widgets/Graph.java

```diff
@@ -977,7 +941,7 @@ void addNode(GraphContainer graphContainer) {
 	}
 
 	void registerItem(GraphItem item) {
-		if (item.getItemType() == GraphItem.NODE) {
+		if (item instanceof GraphNode) {
```

Comment:
not sure if this change is save. Would prefere if we could keep to original version.

Reply:
Or did you combine here the container and the node as both have the same code?

Reply:
but even then I think I would combine both of the original ifs